mlrun/runtimes/databricks_runtime_wrapper.py

```diff
@@ -0,0 +1,110 @@
+    import json
+    import uuid
+
+    from databricks.sdk import WorkspaceClient
+    from databricks.sdk.service.compute import ClusterSpec
+    from databricks.sdk.service.jobs import Run, SparkPythonTask, SubmitTask
+
+    import mlrun
+
+    logger = context.logger
+
+    def upload_file(workspace: WorkspaceClient, script_path_on_dbfs: str):
+        modified_code = internal_code
+        with workspace.dbfs.open(script_path_on_dbfs, write=True, overwrite=True) as f:
+            f.write(modified_code.encode("UTF8"))
+
+    workspace = WorkspaceClient(token=mlrun.get_secret_or_env(key=token_key))
+
+    now = datetime.datetime.now()
+    formatted_date_time = now.strftime("%Y-%m-%d_%H-%M-%S")
+    script_path_on_dbfs = (
+        f"/home/{workspace.current_user.me().user_name}/mlrun_databricks_runtime/"
+        f"sample_{formatted_date_time}_{uuid.uuid4()}.py"
+    )
+
+    upload_file(
+        workspace=workspace,
+        script_path_on_dbfs=script_path_on_dbfs,
+    )
+
+    def print_status(run: Run):
+        statuses = [f"{t.task_key}: {t.state.life_cycle_state}" for t in run.tasks]
+        logger.info(f'workflow intermediate status: {", ".join(statuses)}')
+
+    try:
+        cluster_id = mlrun.get_secret_or_env("DATABRICKS_CLUSTER_ID")
+        if cluster_id:
+            logger.info(f"run with exists cluster_id: {cluster_id}")
+            waiter = workspace.jobs.submit(
+                run_name=f"py-sdk-run-{formatted_date_time}",
+                tasks=[
+                    SubmitTask(
+                        task_key=f"hello_world-{formatted_date_time}",
+                        existing_cluster_id=cluster_id,
+                        spark_python_task=SparkPythonTask(
+                            python_file=f"dbfs:{script_path_on_dbfs}",
+                            parameters=[json.dumps(kwargs)],
+                        ),
+                    )
+                ],
+            )
+        else:
+            logger.info("run with new cluster_id")
+            waiter = workspace.jobs.submit(
+                run_name=f"py-sdk-run-{formatted_date_time}",
+                tasks=[
+                    SubmitTask(
+                        task_key=f"hello_world-{formatted_date_time}",
+                        new_cluster=ClusterSpec(
+                            spark_version=workspace.clusters.select_spark_version(
+                                long_term_support=True
+                            ),
+                            node_type_id=workspace.clusters.select_node_type(
+                                local_disk=True
+                            ),
+                            num_workers=1,
+                        ),
+                        spark_python_task=SparkPythonTask(
+                            python_file=f"dbfs:{script_path_on_dbfs}",
+                            parameters=[json.dumps(kwargs)],
+                        ),
+                    )
+                ],
+            )
```

Comment:
Too duplication, me no likey. How about:
```suggestion
        submit_task_kwargs = {}
        if cluster_id:
            logger.info(f"run with exists cluster_id: {cluster_id}")
            submit_task_kwargs["existing_cluster_id"] = cluster_id
        else:
            logger.info("run with new cluster_id")
            submit_task_kwargs["new_cluster"] = ClusterSpec(
                spark_version=workspace.clusters.select_spark_version(
                    long_term_support=True
                ),
                node_type_id=workspace.clusters.select_node_type(local_disk=True),
                num_workers=1,
            )
        waiter = workspace.jobs.submit(
            run_name=f"py-sdk-run-{formatted_date_time}",
            tasks=[
                SubmitTask(
                    task_key=f"hello_world-{formatted_date_time}",
                    spark_python_task=SparkPythonTask(
                        python_file=f"dbfs:{script_path_on_dbfs}",
                        parameters=[json.dumps(kwargs)],
                    ),
                    **submit_task_kwargs,
                )
            ],
        )
```